lib/view_model/contact_list/contact_list_view_model.dart

```diff
@@ -44,4 +45,20 @@ abstract class ContactListViewModelBase with Store {
   StreamSubscription<BoxEvent>? _subscription;
 
   Future<void> delete(ContactRecord contact) async => contact.original.delete();
+
+  List<ContactRecord> getContacts(CryptoCurrency? cur) {
+    if (cur != null) {
+      return contacts.where((element) => element.type == cur).toList();
+    }
+
+    return contacts;
+  }
+
+  List<WalletContact> getWallets(CryptoCurrency? cur) {
+    if (cur != null) {
+      return walletContacts.where((element) => element.type == cur).toList();
+    }
+
+    return walletContacts;
```

Comment:
let's add the CryptoCurrency to the view model itself and pass it to the view model instead of the page, and we can then make `getContacts` and `getWallets` as getters and annotate them with `@computed` and they would return all the contacts if the currency is null
i.e
`List<ContactRecord> get getContacts => contacts.where((element) => currency == null || element.type == currency).toList();`

Reply:
@OmarHatem28 I did not find easy way to pass `Currency` to the view model because it is passed to the `ContactListPage `from the `router` depending on where it is called.

Reply:
why? we can pass it from the `ContactListPage` factory to the `SendViewModel` like this
`getIt.registerFactoryParam<ContactListPage, CryptoCurrency?, void>((CryptoCurrency? cur, _)
      => ContactListPage(getIt.get<ContactListViewModel>(param1: cur)));`